### ISSUE TITLE
SV: server

### ISSUE THREAD

**Issue by vitiral** [vitiral/artifact#85]
As part of the 1.0 release, the server functionality will be hidden behind a feature flag and development will continue in the "server" branch. 

The `export` cmd will still be supported for generating static html. 

The server will not necessarily compile anymore if the mainline repo. Documentation on how to compile the server will be added to the installation guide. 

This will prevent backwards incompatible changes and thrashing to mainline. Any core changes that are needed for server development will have to be done to mainline through a PR. 

Once the api is complete on the server, it will be merged into mainline so that frontend work can be done without forking. 

# Tasks
- [x] API Calls:
  - [x] DeleteArtifacts
  - [x] CreateArtifacts
  - [x] UpdateArtifacts
  - [x] ReadArtifacts -> ReadProject: we need more information from the server than just the artifacts. For changing things it will go through the above, but for reading you will always get the whole project. (we need at least the files)
- [x] Web UI tasks:
  - [x] Create functionality
  - [x] Edit functionality
  - [x] Delete
  - [x] ~~Enable sorting of completed/tested columns~~ #146
  - [x] add "log list" under nav-bar (user manually clears them for now)
  - [x] add link to defined attribute
  - [x] ~~popup on delete~~ #147
- [x] `art serve` fixes
  - [x] make `-e url` and `-r url` flags. `-e` will put you in an editable mode at the url and will have a warning that it is exteremely unsafe
  - [x] make `-p` flag to specify an alternate port on localhost. By default `art serve` will make an editable instance on local host at port `5373` (SETS)
  - [x] handle `SIGINT` gracefully

**Comment by einsweniger** [comment]
I think I see a CRUD pattern for the API, but `ReadArtifact` is missing, is that omission deliberate?

**Comment by vitiral** [comment]
I make everything plural and I named it GetArtifacts. I would be fine renaming to ReadArtifacts

**Comment by vitiral** [comment]
@einsweniger the above change also changed `GetArtifacts` -> `ReadArtifacts` 😄
